UIViewController-DisplayChild/UIViewController+DisplayChild.swift

```diff
@@ -2,8 +2,9 @@
 
 import UIKit
 
-public protocol InstantiableViewController where Self: UIViewController {
+public protocol Instantiable {
     static func makeInstance() -> Self
+    init()
```

Comment:
Why is the `init()` part of the protocol declaration?

Reply:
Honestly - I don't know. It's not straightforward implementation, it's tricky and you have to know the story behind. The function `makeInstance() -> Self` is from my point of view all you need. It says that you can create an instance with it. So have `init()` there too is confusing.

Reply:
I don't have better solution. If you don't have some, we can resolve this conversation.

Reply:
You are right that it looks strange and unneccessary. Maybe if we put the described helper extension to the lib it might be helpful and it would bring the context aboard. What do you think?

Reply:
The extension example:
```swift
extension Instantiable {
    /// Instantiates entity (especially controller, but can be whatever) from storyboard.
    /// If no suitable storyboard found, the entity is instantiated via a simple init
    /// - example:
    /// `let myViewController = MyViewController.makeInstance()`
    /// - important:
    /// Initial controller of the same type must exists in storyboard named as controller's
    /// class without "ViewController" suffix, otherwise it is instantiated with simple `init()`.
    /// - Returns: Instantiated view controller.
    public static func makeInstance() -> Self {
        let viewControllerName = String(describing: self).replacingOccurrences(of: "ViewController", with: "")
        if Bundle.main.hasStoryboard(name: viewControllerName) {
            let storyboard = UIStoryboard(name: viewControllerName, bundle: nil)
            guard let instance = storyboard.instantiateInitialViewController() as? Self else { fatalError() }
            return instance
        } else {
            return Self()
        }
    }
}

extension UIViewController: Instantiable { }

private extension Bundle {
    //https://stackoverflow.com/questions/38590288/how-to-check-if-a-storyboard-exists-in-a-specific-nsbundle
    func hasStoryboard(name: String) -> Bool {
        return path(forResource: name, ofType: "storyboardc") != nil
    }
}```

Reply:
works for me